server/chat-plugins/scavenger-games.ts

```diff
@@ -451,6 +451,53 @@ const TWISTS: { [k: string]: Twist } = {
 		},
 	},
 
+	pointless: {
+		id: 'pointless',
+		name: 'Pointless',
+		desc: 'Players get bonus points for guessing the least commonly guessed answers.',
+		onAfterLoad() {
+			this.correct = this.questions.map(() => ({}));
+		},
+
+		onCorrectAnswer(player: ScavengerHuntPlayer, value: string) {
+			const curr = player.currentQuestion;
+
+			if (!this.correct[curr][value]) this.correct[curr][value] = [];
+			if (this.correct[curr][value].includes(player.id)) return;
+
+			this.correct[curr][value].push(player.id);
+		},
+
+		onAfterEnd(isReset) {
+			if (isReset) return;
+
+			const buffer = [];
+
+			for (const [idx, data] of this.correct.entries()) {
+				// collect the data for each question
+				const collection = [];
+
+				for (const str in data) {
+					collection.push({ players: data[str], count: data[str].length, value: str });
+				}
+
+				const minValue = collection.reduce((prev, next) => prev.count > next.count ? next : prev).count;
+
+				const matches = collection.filter(entry => entry.count === minValue)
+					.map(entry => [entry.value]).reduce((prev, next) => prev.concat(next));
+				const matchedPlayers = collection.filter(entry => entry.count === minValue)
+					.map(entry => entry.players).reduce((prev, next) => prev.concat(next));
+
+				// display the data
+				const matchDisplay = matches.join(', ');
+				const playerDisplay = `- ${matchedPlayers.join(', ')}`;
+				buffer.push(`Q${idx + 1}: ${matchDisplay} ${playerDisplay}`);
```

Comment:
```suggestion
				const list = Object.entries<string[]>(data).map(([value, players]) => ({ players, count: players.length, value }));
				const minValue = Math.min(...list.map(entry => entry.count));

				const minEntries = list.filter(entry => entry.count === minValue);
				const matchDisplay = minEntries.map(entry => entry.value).join(', ');
				const playerDisplay = minEntries.flatMap(entry => entry.players).join(', ');
				buffer.push(`Q${idx + 1}: ${matchDisplay} - ${playerDisplay}`);
```

Reply:
Slightly overcomplicated with reducer logic, try using something like this onegai